tests/bal_business_logic_suite.py

```diff
@@ -672,9 +672,20 @@ def test_when_bal_library_dir_used_then_expanded_to_library_directory(self):
         data = self.__resolve_to_dict()
         self.assertEqual(data["bal_library_dir"], f"Library is in {expected_dir}")
 
+    def test_when_file_prefixed_but_not_url_then_regurgitated_unmodified(self):
+        data = self.__resolve_to_dict()
+        self.assertEqual(data["a_file_prefixed_non_url"], "file:\\")
+
+    def test_when_raw_file_url_used_then_regurgitated_unmodified(self):
+        data = self.__resolve_to_dict()
+        self.assertEqual(data["a_raw_posix_file_url"], "file:///mnt/per%2520cent")
```

Comment:
Nearly. Before the fix, the URL coming out of BAL would be escaped once, so would become like (1).

If you then passed that string into `FileUrlPathConverter().pathFromUrl`, it would escape again, resulting in (2), which would be wrong.

Essentially the test you highlight assumes a file on disk of `/mnt/per%20cent`.

For background, the actual case where we saw this bug was to do with frame substitution tokens in the string, which are substituted by the host application with a frame number to form the final file name.  The particular format that the host application uses looks a lot like a %-escaped URL token.

Reply:
Gotcha, thanks for the explanation!